src/hotspot/share/utilities/concurrentHashTable.inline.hpp

```diff
@@ -997,7 +997,7 @@ inline size_t ConcurrentHashTable<CONFIG, F>::
       if (dels < num_del) {
         ndel[dels] = rem_n;
       } else {
-        guarantee(dels < std::numeric_limits<int>::max(),
+        guarantee(dels < static_cast<size_t>(std::numeric_limits<int>::max()),
```

Comment:
Does the problem go away if this uses `INT_MAX` instead?

Reply:
Verified on linux-aarch64 that this also fixes this issue. I searched the hotspot shared code and witnessed similar usages. I guess it's better to align with those cases. Fixed.